scripts/harikiri_sqs.py

```diff
@@ -0,0 +1,357 @@
+
+
+DAY_DIR_RE = re.compile(r"jobs/\d{4}/\d{2}/\d{2}/\d{2}/\d{2}$")
+
+NO_JOBS_TIMER = None
+
+KEEP_ALIVE = False
+
+
+def log_event(url, event_type, event_status, event, tags):
+    """Log custom event."""
+
+    params = {
+        "type": event_type,
+        "status": event_status,
+        "event": event,
+        "tags": tags,
+        "hostname": socket.getfqdn(),
+    }
+    headers = {"Content-type": "application/json"}
+    r = requests.post(
+        "%s/event/add" % url, data=json.dumps(params), verify=False, headers=headers
+    )
+    r.raise_for_status()
+    resp = r.json()
+    return resp
+
+
+def keep_alive(root_work):
+    """Check if the keep alive signal exists."""
+
+    return True if os.path.exists(os.path.join(root_work, ".harikiri")) else False
+
+
+def is_jobless(root_work, inactivity_secs, logger=None):
+    """Check if no jobs are running and hasn't run in the past 
+       amount of time passed in.
+    """
+
+    global NO_JOBS_TIMER
+    global KEEP_ALIVE
+
+    # check if keep-alive
+    logging.info("KEEP_ALIVE: %s" % KEEP_ALIVE)
+    if keep_alive(root_work):
+        if KEEP_ALIVE != True:
+            KEEP_ALIVE = True
+            if logger is not None:
+                try:
+                    print((log_event(logger, "harikiri", "keep_alive_set", {}, [])))
+                except:
+                    pass
+        logging.info("Keep-alive exists.")
+        return
+    else:
+        if KEEP_ALIVE != False:
+            KEEP_ALIVE = False
+            if logger is not None:
+                try:
+                    print((log_event(logger, "harikiri", "keep_alive_unset", {}, [])))
+                except:
+                    pass
+            logging.info("Keep-alive removed.")
+
+    most_recent = None
+    for root, dirs, files in os.walk(root_work, followlinks=True):
+        match = DAY_DIR_RE.search(root)
+        if not match:
+            continue
+        dirs.sort()
+        for d in dirs:
+            job_dir = os.path.join(root, d)
+            done_file = os.path.join(job_dir, ".done")
+            if not os.path.exists(done_file):
+                logging.info("%s: no .done file found. Not jobless yet." % job_dir)
+                return False
+            t = os.path.getmtime(done_file)
+            done_dt = datetime.fromtimestamp(t)
+            age = (datetime.utcnow() - done_dt).total_seconds()
+            if most_recent is None or age < most_recent:
+                most_recent = age
+            logging.info("%s: age=%s" % (job_dir, age))
+    if most_recent is None:
+        if NO_JOBS_TIMER is None:
+            NO_JOBS_TIMER = time.time()
+        else:
+            if (time.time() - NO_JOBS_TIMER) > inactivity_secs:
+                return True
+        return False
+    if most_recent > inactivity_secs:
+        return True
+    return False
+
+
+@backoff.on_exception(backoff.expo, ClientError, max_tries=10, max_value=512)
+def get_all_groups(c):
```

Comment:
@anaerobia  can you add a description for the variable `c`?

Reply:
comment added